aimmo-game/simulation/world_map.py

```diff
@@ -87,66 +101,62 @@ class WorldMap(object):
     """
     The non-player world state.
     """
-
     def __init__(self, grid):
-        self.grid = grid
+        self._grid = grid
 
+    @property
+    def num_rows(self):
+        return len(self._grid[0])
+
+    @property
+    def num_cols(self):
+        return len(self._grid)
+
+    @property
+    def num_cells(self):
+        return self.num_rows * self.num_cols
+
+    @property
     def all_cells(self):
-        return (cell for sublist in self.grid for cell in sublist)
+        return (cell for row in self._grid for cell in row)
 
+    @property
     def score_cells(self):
-        return (c for c in self.all_cells() if c.generates_score)
-
-    def potential_spawn_locations(self):
-        return (c for c in self.all_cells()
-                if c.habitable and not c.generates_score and not c.avatar and
-                not c.pickup)
+        return (cell for cell in self.all_cells if cell.generates_score)
 
+    @property
     def pickup_cells(self):
-        return (c for c in self.all_cells() if c.pickup)
-
-    def is_on_map(self, location):
-        return (0 <= location.y < self.num_rows) and (0 <= location.x < self.num_cols)
+        return (cell for cell in self.all_cells if cell.pickup is not None)
 
     def get_cell(self, location):
-        if not self.is_on_map(location):
+        if not self.cell_on_map(location):
             raise ValueError('Location %s is not on the map' % location)
-        cell = self.grid[location.x][location.y]
+        cell = self._grid[location.x][location.y]
         assert cell.location == location, 'location lookup mismatch: arg={}, found={}'.format(location, cell.location)
         return cell
 
-    def clear_cell_actions(self, location):
+    def cell_on_map(self, location):
+        return (0 <= location.y < self.num_rows) and (0 <= location.x < self.num_cols)
+
+    def cell_habitable(self, location):
         try:
-            cell = self.get_cell(location)
-            cell.actions = []
+            return self.get_cell(location).is_habitable
         except ValueError:
-            return
-
-    @property
-    def num_rows(self):
-        return len(self.grid[0])
-
-    @property
-    def num_cols(self):
-        return len(self.grid)
-
-    @property
-    def num_cells(self):
-        return self.num_rows * self.num_cols
+            return False
 
-    def apply_score(self):
-        for cell in self.score_cells():
-            try:
-                cell.avatar.score += 1
-            except AttributeError:
-                pass
+    def cell_occupied(self, location):
```

Comment:
This is never called by anything. I don't think we need to replicate the `Cell` interface here as you can always access that using `get_cell`.